File: python/hello_debugger.py
#!/usr/bin/env python

from argparse import ArgumentParser
from hello import say_hi


def parse_args():
    parser = ArgumentParser(
        prog="Hello Debugger",
        description="A Simple hello world app to test vscode debugging",
    )
    parser.add_argument("-n", "--name", help="who to say hi to", required=True)
    args = parser.parse_args()
    return args


if __name__ == "__main__":
    args = parse_args()
    say_hi(args.name)  # <------------- put a breakpoint here!

File: typescript/src/hello-debugger.ts
#!/usr/bin/env ts-node

import yargs from 'yargs';

const argv = yargs.alias("n", "name").nargs('n', 1).require('name', "Name Required").argv;

(() => {
   const name = argv.name;
   console.log(`Hello, ${name}`);
})();
